### ISSUE TITLE
Use HParams for "learning rate" hyper-parameter tuning

### ISSUE THREAD

**Issue by yashmanuda** [tensorflow/tensorflow#34275]
https://www.tensorflow.org/tensorboard/hyperparameter_tuning_with_hparams

## Description of issue (what needs changing):
In the documentation it's nowhere written how to tune learning rate hyper-parameter for different optimisation algorithms. Plus the code in file **python3.6/site-packages/tensorboard/plugins/hparams/summary_v2.py** says,
`if dtype not in (int, float, bool, str):
--> 482       raise ValueError("Unknown dtype: %r" % (dtype,))` 

I am trying to create HParam list 

`hp_optimizer = hp.HParam('optimizer', hp.Discrete([optimizers.Adam(learning_rate=0.001),
                                                   optimizers.Adam(learning_rate=0.0001),
                                                   optimizers.Adamax(learning_rate=0.001),
                                                   optimizers.Adamax(learning_rate=0.0001),
                                                   optimizers.SGD(learning_rate=0.01),
                                                   optimizers.SGD(learning_rate=0.001)]))`

### Clear description
How should someone tune learning rate?

**Comment by ravikyram** [comment]
@yashmanuda 
Will it be possible to provide minimal stand alone code to reproduce the issue. It helps us in localizing the issue faster.Thanks!

**Comment by jtavrisov** [comment]
There's no functionality asfaik to do the tuning for you. This is just a dashboard so you can compare different runs with different hparams to see which is best in tensorboard.

I have a GridSearch/RandomSearch function in my code that just loops over parameters and runs a new experiment. Just a more automated fashion of running different trials.

As for your error, you are passing in an instance of Adam as an hparam, only numbers/strings are allowed. If you want learning rate to be displayed in the Hparams tab then you should pass in just learning rate. I personally add optimizer as a string and then learning rate as a float.

Your link as plenty of examples of all of this

**Comment by ravikyram** [comment]
@yashmanuda 

Any update on this issue please. Thanks!

**Comment by yashmanuda** [comment]
I am able to use float as the learning rate, and using it to create an optimizer. Thanks for the help.

**Comment by ravikyram** [comment]
@yashmanuda 

Looks like issue got resolved.Please, let us know whether we can close this issue?. Thanks!

**Comment by ravikyram** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!

**Comment by arwen-x** [comment]
> There's no functionality asfaik to do the tuning for you. This is just a dashboard so you can compare different runs with different hparams to see which is best in tensorboard.
> 
> I have a GridSearch/RandomSearch function in my code that just loops over parameters and runs a new experiment. Just a more automated fashion of running different trials.
> 
> As for your error, you are passing in an instance of Adam as an hparam, only numbers/strings are allowed. If you want learning rate to be displayed in the Hparams tab then you should pass in just learning rate. I personally add optimizer as a string and then learning rate as a float.
> 
> Your link as plenty of examples of all of this

Could you please give more details on adding optimizer as a string and learning rate as a float?I am not able to experiment with different learning rates on HParams. Here is the [link](https://github.com/tensorflow/tensorboard/issues/3688) to my issue.
